cli/command/utils.go

```diff
@@ -90,9 +95,40 @@ func PromptForConfirmation(ins io.Reader, outs io.Writer, message string) bool {
 		ins = streams.NewIn(os.Stdin)
 	}
 
-	reader := bufio.NewReader(ins)
-	answer, _, _ := reader.ReadLine()
-	return strings.ToLower(string(answer)) == "y"
+	result := make(chan bool)
+	defer close(result)
+
+	// We catch the termination signal
+	// and exit the prompt gracefully
+	// it is up to the command to handle the termination
+	notifyCtx, notifyCancel := signal.NotifyContext(ctx, syscall.SIGINT, syscall.SIGTERM)
+	defer notifyCancel()
+
+	go func() {
+		defer notifyCancel()
+		scanner := bufio.NewScanner(ins)
+		for scanner.Scan() {
```

Comment:
2 things here:

- We don't need a for loop here, since in both of the `select` cases we return regardless.
- `scanner.Scan()` is still a blocking call, so if the user presses CTRL-C without sending any other input, this goroutine should still be blocked on the Scan(). If we ever have multiple confirmation prompts, the first prompt's goroutine will still be running, it will capture user input from the second confirmation prompt, then it will panic trying to send the result down the closed `result` channel

IMHO this second point warrants some test cases, where we basically just call this function a few times in a row with the various user input possibilities (the permutations of the `y/Y/n/N/CTRL-C` options available). It could become confusing for the user if we use multiple confirmations and the CLI doesn't behave as expected